### PR TITLE
Update GT in CMSSW to 105X for ultra legacy

### PR DESCRIPTION
**Pull request by tocheng**
Hello,

The PR is to migrate the GTs in CMSSW release (autoCond) to **105X GTs**,
and includes the most update data/MC conditions for ultra legacy preparation.
In addition, new GEM emap, PPS timing, LHCInfo are included in data GTs.

### **Offline data GT for run1 and run2**
The updated GT is
https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/105X_dataRun2_v6
Diff :
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/105X_dataRun2_v6/103X_dataRun2_v6

- New GEM eMap to support #25158 for new unpacker
- New PPS timing and LHCInfo to support #25495 for PPS proton reconstruction
(- New PPS timing correction to support #25858 to use timing in PPS local reconstruction)
- New HCAL data condition announced in AlCa-hn :
   https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/4075.html

### Offline relval data GT
The updated GT is
https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/105X_dataRun2_relval_v5
Diff :
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/105X_dataRun2_relval_v5/103X_dataRun2_relval_v10

- ECAL pedestal changes from prompt to offline
- New GEM eMap to support #25158 for new unpacker
- New PPS timing and LHCInfo to support #25495 for PPS proton reconstruction
(- New PPS timing correction to support #25858 to use timing in PPS local reconstruction)
- Updated HCAL data condition announced in AlCa-hn :
   https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/4075.html

### Prompt like data GT
The updated GT is
https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/105X_dataRun2_PromptLike_HEfail_v3
https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/105X_dataRun2_PromptLike_v5
Diff :
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/105X_dataRun2_PromptLike_HEfail_v3/103X_dataRun2_PromptLike_HEfail_v5
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/105X_dataRun2_PromptLike_v5/103X_dataRun2_PromptLike_v6

- New GEM eMap to support #25158 for new unpacker
- New PPS timing and LHCInfo to support #25495 for PPS proton reconstruction
(- New PPS timing correction to support #25858 to use timing in PPS local reconstruction)

### **2017 Realistic GT**
The updated GT is
https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/105X_mc2017_realistic_v4
Diff :
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/105X_mc2017_realistic_v4/103X_mc2017_realistic_v2

- Update ECAL condition and PF rechit threshold. 
   https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/4085.html
- Updated HCAL MC condition announced in AlCa-hn :
   https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/4075.html
- Update pixel  local reconstruction conditions
- Update pixel  2D template for cluster charge reweighting

### **2017 Design GT**
The updated GT is
https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/105X_mc2017_design_IdealBS_v3
Diff :
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/105X_mc2017_design_IdealBS_v3/103X_mc2017_design_IdealBS_v2

- Update ES inter-calibration/pedestal. See talk at AlCaDB meeting 
   https://indico.cern.ch/event/791536/contributions/3288530/attachments/1782158/2899777/AlCaDB_21_01_2019.pdf
- Updated HCAL MC condition announced in AlCa-hn :
   https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/4075.html

### **2017 Cosmic MC GT**
The updated GT is
https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/105X_mc2017cosmics_realistic_peak_v5
https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/105X_mc2017cosmics_realistic_deco_v4
Diff :
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/105X_mc2017cosmics_realistic_peak_v5/103X_mc2017cosmics_realistic_peak_v2
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/105X_mc2017cosmics_realistic_deco_v4/103X_mc2017cosmics_realistic_deco_v2

- Update ECAL condition and PF rechit threshold. 
   https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/4085.html
- Updated HCAL MC condition announced in AlCa-hn :
   https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/4075.html
- Update pixel  local reconstruction conditions
- Update pixel  2D template for cluster charge reweighting

### **2018 Realistic GT (with HE fail)**
The updated GT is
https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/105X_upgrade2018_realistic_v2
https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/105X_upgrade2018_realistic_HEfail_v3
Diff :
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/105X_upgrade2018_realistic_v2/103X_upgrade2018_realistic_v8
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/105X_upgrade2018_realistic_HEfail_v3/103X_upgrade2018_realistic_HEfail_v4

- Update ES inter-calibration/pedestal. See talk at AlCaDB meeting 
   https://indico.cern.ch/event/791536/contributions/3288530/attachments/1782158/2899777/AlCaDB_21_01_2019.pdf
- Updated HCAL MC condition announced in AlCa-hn :
   https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/4075.html
- Update pixel  local reconstruction conditions
- Update pixel  2D template for cluster charge reweighting
- New labelled pixel quality to support new digitizer #25885 

### **2018 Design GT**
The updated GT is
https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/105X_upgrade2018_design_v2
Diff :
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/105X_upgrade2018_design_v2/103X_upgrade2018_design_v4

- Update ES inter-calibration/pedestal. See talk at AlCaDB meeting 
   https://indico.cern.ch/event/791536/contributions/3288530/attachments/1782158/2899777/AlCaDB_21_01_2019.pdf
- Updated HCAL MC condition announced in AlCa-hn :
   https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/4075.html
- New labelled pixel quality to support new digitizer #25885 

### **2018 Cosmic GT**
The updated GT is
https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/105X_upgrade2018cosmics_realistic_peak_v3
https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/105X_upgrade2018cosmics_realistic_deco_v3
Diff :
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/105X_upgrade2018cosmics_realistic_peak_v3/103X_upgrade2018cosmics_realistic_peak_v1
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/105X_upgrade2018cosmics_realistic_deco_v3/103X_upgrade2018cosmics_realistic_deco_v7

- Update ES inter-calibration/pedestal. See talk at AlCaDB meeting 
   https://indico.cern.ch/event/791536/contributions/3288530/attachments/1782158/2899777/AlCaDB_21_01_2019.pdf
- Updated HCAL MC condition announced in AlCa-hn :
   https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/4075.html
- Update pixel  local reconstruction conditions
- Update pixel  2D template for cluster charge reweighting
- New labelled pixel quality to support new digitizer #25885 

### **2016 Realistic GT**
The updated GT is
https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/105X_mcRun2_asymptotic_v1
Diff :
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/105X_mcRun2_asymptotic_v1/103X_mcRun2_asymptotic_v3

- Update ES inter-calibration/pedestal. See talk at AlCaDB meeting 
   https://indico.cern.ch/event/791536/contributions/3288530/attachments/1782158/2899777/AlCaDB_21_01_2019.pdf
- Updated HCAL MC condition announced in AlCa-hn :
   https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/4075.html

### **2016 MC Design GT**
The updated GT is
https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/105X_mcRun2cosmics_startup_deco_v1
Diff :
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/105X_mcRun2cosmics_startup_deco_v1/103X_mcRun2_design_v3
- Updated HCAL data condition announced in AlCa-hn :
   https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/4075.html

### **2016 Cosmic MC GT**
The updated GT is
https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/105X_mcRun2_design_v1
Diff :
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/105X_mcRun2cosmics_startup_deco_v1/103X_mcRun2cosmics_startup_deco_v3
- Update ES inter-calibration/pedestal. See talk at AlCaDB meeting 
   https://indico.cern.ch/event/791536/contributions/3288530/attachments/1782158/2899777/AlCaDB_21_01_2019.pdf
- Updated HCAL data condition announced in AlCa-hn :
   https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/4075.html

### **Run-3 MC Realistic GT**
The updated GT is
https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/105X_postLS2_realistic_v3
Diff : 
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/105X_postLS2_realistic_v3/103X_postLS2_realistic_v4
- include labelled pixel quality to support digitizer development #25885 

### **Run-3 MC Design GT**
The updated GT is
https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/105X_postLS2_design_v1
Diff : 
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/105X_postLS2_design_v1/103X_postLS2_design_v4
- include labelled pixel quality to support digitizer development #25885 

### **Phase-2 GT**
The updated GT is
https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/105X_upgrade2023_realistic_v2
Diff : 
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/105X_upgrade2023_realistic_v2/103X_upgrade2023_realistic_v2
- include labelled pixel quality to support digitizer development #25885

### CODE REVIEW

**Comment by tocheng** [comment]
@cmsbuild please test

**Comment by mmusich** [comment]
@tocheng will this update take into account all the requested changes needed for #25885 to work?

**Comment by boudoul** [comment]
Hello @tocheng - would it be possible for Alca to reintroduce the good habits they had until recently which was to make explicit all the changes in the description of the pull request , otherwise it will be really impossible to keep track and also for the validators to undertand what needs to be validated : I give you here just an example of what alca was routinely doing in the recent past for every PR updating GTs , which was really perfect  : 
https://github.com/cms-sw/cmssw/pull/23619
Thank you

**Comment by tocheng** [comment]
> @tocheng will this update take into account all the requested changes needed for #25885 to work?

@mmusich For MC GT >=2018, there is a labelled pixel quality with the same tag as the unlabelled one.

**Comment by tocheng** [comment]
@cmsbuild please test

**Comment by tocheng** [comment]
@cmsbuild please abort

File: Configuration/AlCa/python/autoCond.py [hunk 1]
```diff
@@ -44,31 +44,31 @@
...
+    'phase1_2018_cosmics'      :   '105X_upgrade2018cosmics_realistic_deco_v3',
     # GlobalTag for MC production (cosmics) with realistic conditions for full Phase1 2018 detector,  Strip tracker in PEAK mode
-    'phase1_2018_cosmics_peak' :   '103X_upgrade2018cosmics_realistic_peak_v1',
+    'phase1_2018_cosmics_peak' :   '105X_upgrade2018cosmics_realistic_peak_v3',
     # GlobalTag for MC production with perfectly aligned and calibrated detector for Phase1 2019
-    'phase1_2019_design'       : '103X_postLS2_design_v4', # GT containing design conditions for postLS2
+    'phase1_2019_design'       : '105X_postLS2_design_v1', # GT containing design conditions for postLS2
     # GlobalTag for MC production with perfectly aligned and calibrated detector for Phase1 2019
-    'phase1_2019_realistic'    : '103X_postLS2_realistic_v4', # GT containing realistic conditions for postLS2
+    'phase1_2019_realistic'    : '105X_postLS2_realistic_v2', # GT containing realistic conditions for postLS2
```
Comment (mmusich) [inline Configuration/AlCa/python/autoCond.py]: Hi @tocheng, 
I am a trifle confused by the last change about reverting the MC misalignment scenario in the 2019 queue:
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/105X_postLS2_realistic_v1/105X_postLS2_realistic_v2,
`TrackerAlignment_Upgrade2017_realistic_v2`  contains an unintentionally large weak mode and should therefore be removed from all autoCond GTs.
The "spurious changes" I mentioned in https://github.com/cms-sw/cmssw/pull/25885#issuecomment-462333444 were spurious only in the context of PR #25885 which was meant to be technical and not change physics output. As you are changing conditions for real here, I would prefer to keep the fixed alignment. Thanks

Comment (tocheng) [reply Configuration/AlCa/python/autoCond.py]: OK. I agree.

File: Configuration/AlCa/python/autoCond.py [hunk 2]
```diff
@@ -44,31 +44,31 @@
...
     # GlobalTag for MC production (cosmics) with realistic conditions for full Phase1 2018 detector,  Strip tracker in DECO mode
-    'phase1_2018_cosmics'      :   '103X_upgrade2018cosmics_realistic_deco_v7',
+    'phase1_2018_cosmics'      :   '105X_upgrade2018cosmics_realistic_deco_v3',
     # GlobalTag for MC production (cosmics) with realistic conditions for full Phase1 2018 detector,  Strip tracker in PEAK mode
-    'phase1_2018_cosmics_peak' :   '103X_upgrade2018cosmics_realistic_peak_v1',
+    'phase1_2018_cosmics_peak' :   '105X_upgrade2018cosmics_realistic_peak_v3',
     # GlobalTag for MC production with perfectly aligned and calibrated detector for Phase1 2019
-    'phase1_2019_design'       : '103X_postLS2_design_v4', # GT containing design conditions for postLS2
+    'phase1_2019_design'       : '105X_postLS2_design_v1', # GT containing design conditions for postLS2
     # GlobalTag for MC production with perfectly aligned and calibrated detector for Phase1 2019
```
Comment (mmusich) [inline Configuration/AlCa/python/autoCond.py]: BTW sorry for being picky but this comment
```python
# GlobalTag for MC production with perfectly aligned and calibrated detector for Phase1 2019
```
doesn't seem accurate for a realistic GT.

**Comment by tocheng** [comment]
@cmsbuild please test

**Comment by tocheng** [comment]
@cmsbuild please test

**Comment by tocheng** [comment]
@cmsbuild please test

**Comment by tocheng** [comment]
@boudoul @mmusich 
Many thanks for the comments.
AlCa will follow the progress of release and release validation more closely to avoid massive condition updates in one go.

There is one comparison failure in single photon 2016 data workflow.
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_10_5_X_2019-02-13-1100+25929/30418/validateJR/all_OldVSNew_RunSinglePh2016Bwf136p731/
It is because the offline data relval GT uses offline ECAL pedestal instead of prompt

**Comment by boudoul** [comment]
thanks to you !

**Comment by mmusich** [comment]
Hi @tocheng 
are you sure about this:
> Remove unused unlabeled pixel 2D template 

?
Doesn't look to be the case for the offline and offline relval GTs:

```
Record                   Label      pro::105X_dataRun2_relval_v5 Tag               pro::103X_dataRun2_relval_v10 Tag              
-----------------------  ---------  ---------------------------------------------  ---------------------------------------------  
EcalPedestalsRcd         -          EcalPedestals_Run1_Run2_Time_v1                EcalPedestals_prompt                           
GEMELMapRcd              -          GEMELMap_v3                                    GEMELMap_v2                                    
GEMeMapRcd               -          GEMeMap_v3                                     -                                              
HcalChannelQualityRcd    -          HcalChannelQuality_v8.0_offline                HcalChannelQuality_v7.20_offline               
HcalGainsRcd             -          HcalGains_v7.0_offline                         HcalGains_v6.5_offline                         
HcalPedestalWidthsRcd    -          HcalPedestalWidths_ADC_v2.1_offline            HcalPedestalWidths_ADC_v1.2_offline            
HcalPedestalWidthsRcd    effective  HcalPedestalWidths_ADC_v2.1_offline_effective  HcalPedestalWidths_ADC_v1.1_offline_effective  
HcalPedestalsRcd         effective  HcalPedestals_ADC_v9.3_offline_effective       HcalPedestals_ADC_v9.2_offline_effective       
HcalRespCorrsRcd         -          HcalRespCorrs_v7.0_offline                     HcalRespCorrs_v6.5_offline                     
HcalTPParametersRcd      -          HcalTPParameters_v3.0_hlt                      HcalTPParameters_v2.0_offline                  
LHCInfoRcd               -          LHCInfoEndFill_prompt_v0                       -                                              
PPSTimingCalibrationRcd  -          TotemTimingCalibration_v1                      -                      
```
and
```
Record                   Label      pro::105X_dataRun2_v6 Tag                      pro::103X_dataRun2_v6 Tag                      
-----------------------  ---------  ---------------------------------------------  ---------------------------------------------  
GEMELMapRcd              -          GEMELMap_v3                                    GEMELMap_v2                                    
GEMeMapRcd               -          GEMeMap_v3                                     -                                              
HcalChannelQualityRcd    -          HcalChannelQuality_v8.0_offline                HcalChannelQuality_v7.20_offline               
HcalGainsRcd             -          HcalGains_v7.0_offline                         HcalGains_v6.5_offline                         
HcalPedestalWidthsRcd    -          HcalPedestalWidths_ADC_v2.1_offline            HcalPedestalWidths_ADC_v1.2_offline            
HcalPedestalWidthsRcd    effective  HcalPedestalWidths_ADC_v2.1_offline_effective  HcalPedestalWidths_ADC_v1.1_offline_effective  
HcalPedestalsRcd         effective  HcalPedestals_ADC_v9.3_offline_effective       HcalPedestals_ADC_v9.2_offline_effective       
HcalRespCorrsRcd         -          HcalRespCorrs_v7.0_offline                     HcalRespCorrs_v6.5_offline                     
HcalTPParametersRcd      -          HcalTPParameters_v3.0_hlt                      HcalTPParameters_v2.0_offline                  
LHCInfoRcd               -          LHCInfoEndFill_prompt_v0                       -                                              
PPSTimingCalibrationRcd  -          TotemTimingCalibration_v1                      -                            
```

**Comment by tocheng** [comment]
> Hi @tocheng
> are you sure about this:
> 
> > Remove unused unlabeled pixel 2D template
> 
> ?
> Doesn't look to be the case for the offline and offline relval GTs:
> 
> ```
> Record                   Label      pro::105X_dataRun2_relval_v5 Tag               pro::103X_dataRun2_relval_v10 Tag              
> -----------------------  ---------  ---------------------------------------------  ---------------------------------------------  
> EcalPedestalsRcd         -          EcalPedestals_Run1_Run2_Time_v1                EcalPedestals_prompt                           
> GEMELMapRcd              -          GEMELMap_v3                                    GEMELMap_v2                                    
> GEMeMapRcd               -          GEMeMap_v3                                     -                                              
> HcalChannelQualityRcd    -          HcalChannelQuality_v8.0_offline                HcalChannelQuality_v7.20_offline               
> HcalGainsRcd             -          HcalGains_v7.0_offline                         HcalGains_v6.5_offline                         
> HcalPedestalWidthsRcd    -          HcalPedestalWidths_ADC_v2.1_offline            HcalPedestalWidths_ADC_v1.2_offline            
> HcalPedestalWidthsRcd    effective  HcalPedestalWidths_ADC_v2.1_offline_effective  HcalPedestalWidths_ADC_v1.1_offline_effective  
> HcalPedestalsRcd         effective  HcalPedestals_ADC_v9.3_offline_effective       HcalPedestals_ADC_v9.2_offline_effective       
> HcalRespCorrsRcd         -          HcalRespCorrs_v7.0_offline                     HcalRespCorrs_v6.5_offline                     
> HcalTPParametersRcd      -          HcalTPParameters_v3.0_hlt                      HcalTPParameters_v2.0_offline                  
> LHCInfoRcd               -          LHCInfoEndFill_prompt_v0                       -                                              
> PPSTimingCalibrationRcd  -          TotemTimingCalibration_v1                      -                      
> ```
> 
> and
> 
> ```
> Record                   Label      pro::105X_dataRun2_v6 Tag                      pro::103X_dataRun2_v6 Tag                      
> -----------------------  ---------  ---------------------------------------------  ---------------------------------------------  
> GEMELMapRcd              -          GEMELMap_v3                                    GEMELMap_v2                                    
> GEMeMapRcd               -          GEMeMap_v3                                     -                                              
> HcalChannelQualityRcd    -          HcalChannelQuality_v8.0_offline                HcalChannelQuality_v7.20_offline               
> HcalGainsRcd             -          HcalGains_v7.0_offline                         HcalGains_v6.5_offline                         
> HcalPedestalWidthsRcd    -          HcalPedestalWidths_ADC_v2.1_offline            HcalPedestalWidths_ADC_v1.2_offline            
> HcalPedestalWidthsRcd    effective  HcalPedestalWidths_ADC_v2.1_offline_effective  HcalPedestalWidths_ADC_v1.1_offline_effective  
> HcalPedestalsRcd         effective  HcalPedestals_ADC_v9.3_offline_effective       HcalPedestals_ADC_v9.2_offline_effective       
> HcalRespCorrsRcd         -          HcalRespCorrs_v7.0_offline                     HcalRespCorrs_v6.5_offline                     
> HcalTPParametersRcd      -          HcalTPParameters_v3.0_hlt                      HcalTPParameters_v2.0_offline                  
> LHCInfoRcd               -          LHCInfoEndFill_prompt_v0                       -                                              
> PPSTimingCalibrationRcd  -          TotemTimingCalibration_v1                      -                            
> ```

Hello Marco,
you are right. Let me remove the statement.

**Comment by tocheng** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**